src/http/common.cr

```diff
@@ -82,7 +82,22 @@ module HTTP
     end
 
     io << "\r\n"
-    io << body if body
+
+    if body
+      if body.is_a?(IO)
+        if headers["Transfer-Encoding"] == "chunked"
```

Comment:
I think that feels better, yes. Basically just do
```cr
headers["Transfer-Encoding"] ||= "chunked" if body && body.is_a?(IO)
serialize_headers
if body
  if body.is_a?(IO)
    # ...
```

I guess Response should still set `Content-Length` from `body.size || body_io.size` if possible and not set already.